### PR TITLE
[ops] Automate and dedup address config

### PR DESCRIPTION
**Pull request by ujvl**
- dedups address configuration (everything should be bootstrapped from single dotenv, currently `.contracts.env`)
- dedups base_rollup.json params (can be set automatically from genesis.json during config generation)
- removes dead/obsolete code in hardhat cfg and some other places
- cleans up scripts with common util
- adds wallet generation tool for convenience

### CODE REVIEW

File: config/e2e_test/base_sp_rollup.json [hunk 1]
```diff
@@ -1,23 +1,12 @@
...
-        },
-        "l2": {
-            "number": 0
-        },
         "system_config": {
-            "batcherAddr": "0xf39fd6e51aad88f6f4ce6ab8827279cfffb92266",
             "overhead": "0x0000000000000000000000000000000000000000000000000000000000000000",
-            "scalar": "0x0000000000000000000000000000000000000000000000000000000000000000",
-            "gasLimit": 30000000
+            "scalar": "0x0000000000000000000000000000000000000000000000000000000000000000"
```
Comment (LEAFERx) [inline config/e2e_test/base_sp_rollup.json]: maybe change the `overhead` and `scalar` to some non-zero value

File: contracts/hardhat.config.ts [hunk 2]
```diff
@@ -117,29 +67,14 @@ const config: HardhatUserConfig = {
     clear: true,
   },
   namedAccounts: {
-    sequencer: {
```
Comment (LEAFERx) [inline contracts/hardhat.config.ts:120]: why not export sequencer as a named account. if we're using single source of truth we can only set the default to env.

File: ops/cmd/genesis/main.go [hunk 3]
```diff
@@ -54,6 +56,10 @@ var Flags = []cli.Flag{
 		Name:  "l1-block",
 		Usage: "L1 block number",
 	},
+	&cli.StringFlag{
```
Comment (LEAFERx) [inline ops/cmd/genesis/main.go:60]: alloc is not only about balance -- we might also want to set nonce/code/storage sometime later manually

File: ops/cmd/genesis/main.go [hunk 4]
```diff
@@ -86,6 +92,15 @@ func GenerateSpecularGenesis(ctx *cli.Context) error {
 	if err != nil {
 		return err
 	}
+	if ctx.IsSet("extra-allocs") {
```
Comment (LEAFERx) [inline ops/cmd/genesis/main.go]: where is this flag defined?

**Review by LEAFERx — COMMENTED** [review]
Several questions:
- where is sequencer funded on l2 genesis? through `--allocs` flag?
- how can we fund a redeploy like `L2Portal`? will `--allocs` flag's balance value be overwritten by `BuildGenesis`? 
- I think maybe a better way is to still use the `allocs` json field instead of the flag, but allowing predefined replaces like writing genesis config as `{ "sequencerAddr" : { "balance": ... }}` and `sequencerAddr` will be replaced by the actual value from env.

i didn't check scripts and sidecar closely.

File: ops/cmd/genesis/main.go (hunk 3 above)
Comment (ujvl) [reply ops/cmd/genesis/main.go:60]: yeah this is kind of just a hack, we can clean this up later; but also if you want to you can still use `alloc` in the config—this is just for convenience to merge the dotenv vars in

File: ops/cmd/genesis/main.go (hunk 4 above)
Comment (ujvl) [reply ops/cmd/genesis/main.go]: supposed to be `alloc`, fixed

File: contracts/hardhat.config.ts (hunk 2 above)
Comment (ujvl) [reply contracts/hardhat.config.ts:120]: iiuc it depends on the priv key then, which is unnecessary

File: config/e2e_test/base_sp_rollup.json (hunk 1 above)
Comment (ujvl) [reply config/e2e_test/base_sp_rollup.json]: do these values even matter, don't they get overloaded

also moved them to the config generation

**Review by chalkandpaste — APPROVED** [review]

**Pull request merged**